Sources/SwiftProtobuf/Map.swift

```diff
@@ -15,10 +15,26 @@
 import Foundation
 import Swift
 
-public struct ProtobufMap<KeyType: MapKeyType, ValueType: MapValueType>
+public struct ProtobufMap<KeyType: MapKeyType, ValueType: FieldType>
     where KeyType.BaseType: Hashable
 {
-    typealias Key = KeyType.BaseType
-    typealias Value = ValueType.BaseType
+    public typealias Key = KeyType.BaseType
+    public typealias Value = ValueType.BaseType
+    public typealias BaseType = Dictionary<Key, Value>
+}
+
+public struct ProtobufMessageMap<KeyType: MapKeyType, ValueType: Message>
+    where KeyType.BaseType: Hashable
```

Comment:
Since `Message` and `Enum` are no longer `FieldType`, we can't use the generic `ProtobufMap` for them.  So I ended up triplicating most of the map support.